### PR TITLE
Feature/ Add Assessment missing function events

### LINKED ISSUE #424: Assessment: add events for stake/unstake/withdrawRewards/castVote methods

### PR DESCRIPTION
**Pull request by patitonar**
## Context

Fixes #424 

## Changes proposed in this pull request

Adds events to the `stake()`, `unstake()`,  `withdrawRewards()`,  `withdrawRewardsTo()` and `castVote()` methods.

Used the events already defined in `contracts/interfaces/IAssessment.sol`. The only event that was updated is the `VoteCast` event to include the assessment id as a parameter.

## Test plan

Unit tests for the corresponding events were introduced in:
- `test/unit/Assessment/stake.js`
- `test/unit/Assessment/unstake.js`
- `test/unit/Assessment/withdrawRewards.js`
- `test/unit/Assessment/withdrawRewardsTo.js`
- `test/unit/Assessment/castVotes.js`

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: contracts/modules/assessment/Assessment.sol [hunk 1]
```diff
@@ -162,6 +164,8 @@ contract Assessment is IAssessment, MasterAwareV2 {
 
     stakeOf[msg.sender].amount -= amount;
     nxm.transfer(to, amount);
+
+    emit StakeWithdrawn(msg.sender, amount);
```
Comment (danoctavian) [inline contracts/modules/assessment/Assessment.sol:168]: can add the destination address here as well `to`

File: test/unit/Assessment/stake.js [hunk 2]
```diff
@@ -40,4 +40,14 @@ describe('stake', function () {
       assert(balance.eq(parseEther('200')));
     }
   });
+
+  it('emits StakeDeposited event with staker and amount', async function () {
+    const { assessment } = this.contracts;
+    const user = this.accounts.members[0];
```
Comment (danoctavian) [inline test/unit/Assessment/stake.js:46]: ```const [user] = this.accounts.members;```

Looks better as a pattern

File: contracts/interfaces/IAssessment.sol [hunk 3]
```diff
@@ -170,7 +170,7 @@ interface IAssessment {
 
   event StakeDeposited(address user, uint104 amount);
   event StakeWithdrawn(address indexed user, uint96 amount);
-  event VoteCast(address indexed user, uint96 stakedAmount, bool accepted);
+  event VoteCast(address indexed user, uint256 assessmentId, uint96 stakedAmount, bool accepted);
```
Comment (danoctavian) [inline contracts/interfaces/IAssessment.sol:173]: good

File: contracts/modules/assessment/Assessment.sol [hunk 4]
```diff
@@ -237,6 +241,8 @@ contract Assessment is IAssessment, MasterAwareV2 {
     // This is the index where the next withdrawReward call will start iterating from
     stakeOf[staker].rewardsWithdrawableFromIndex = SafeUintCast.toUint104(withdrawnUntilIndex);
     ITokenController(getInternalContractAddress(ID.TC)).mint(destination, withdrawn);
+
+    emit RewardWithdrawn(staker, withdrawn);
```
Comment (patitonar) [inline contracts/modules/assessment/Assessment.sol:245]: @danoctavian Do you think it is also worth adding the destination address to the `RewardWithdrawn` event?

**Review by danoctavian — CHANGES_REQUESTED** [review]
Small changes required

Otherwise looks good

File: contracts/modules/assessment/Assessment.sol (hunk 4 above)
Comment (danoctavian) [reply contracts/modules/assessment/Assessment.sol:245]: Yeah i don't see a downside

**Review by danoctavian — APPROVED** [review]
LGTM

**Pull request merged**